discord_slash/model.py

```diff
@@ -358,3 +379,67 @@ async def wrap():
                     await self._http.delete(self.__interaction_token, self.id)
 
             self._state.loop.create_task(wrap())
+
+
+class PermissionData:
+    """
+    Single slash permission data.
+
+    :ivar id: User or role id, based on following type specfic.
+    :ivar type: The ``SlashCommandPermissionsType`` type of this permission.
+    :ivar permission: State of permission. ``True`` to allow, ``False`` to disallow.
+    """
+    def __init__(self, id, type, permission, **kwargs):
+        self.id = id
+        self.type = type
+        self.permission = permission
+
+    def __eq__(self, other):
+        if isinstance(other, PermissionData):
+            return (
+                self.id == other.id
+                and self.type == other.id
+                and self.permission == other.permission
+            )
+        else:
+            return False
+
+
+class GuildPermissionsData:
+    """
+    Slash permissions data for a command in a guild.
+
+    :ivar id: Command id, provided by discord.
+    :ivar guild_id: Guild id that the permissions are in. 
+    :ivar permissions: List of permissions dict.
+    """
+    def __init__(self, id, guild_id, permissions, **kwargs):
+        self.id = id
+        self.guild_id = guild_id
+        self.permissions = []
+        if permissions:
+            for permission in permissions:
+                self.permissions.append(PermissionData(**permission))
+
+    def __eq__(self, other):
+        if isinstance(other, GuildPermissionsData):
+            return (
+                self.id == other.id
+                and self.guild_id == other.guild_id
+                and self.permissions == other.permissions
+            )
+        else:
+            return False
+
+
+class SlashCommandPermissionsType(IntEnum):
```

Comment:
```suggestion
class SlashCommandPermissionType(IntEnum):
```

It's for a single permission, and discord uses singular 'permission'.